### PR TITLE
[BEAM-12024] Move examples.wordcount_dataframe to examples.dataframe, add taxiride example

### PR DESCRIPTION
**Pull request by TheNeuralBit**
This PR creates a new `examples.dataframe` module and moves `wordcount_dataframe` into it. It also creates two new example pipelines in `examples.dataframe.taxiride`, one based on the aggregation pipeline from https://beam.apache.org/blog/dataframe-api-preview-available/, and one that augments that pipeline with a join on the taxi zone lookup table.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/apache_beam/examples/dataframe/taxiride_test.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+  def test_aggregation(self):
+    # Compute expected result
+    rides = pd.read_csv(self.input_path)
+    expected_counts = rides.groupby('DOLocationID').passenger_count.sum()
+
+    taxiride.run_aggregation_pipeline([], self.input_path, self.output_path)
+
+    # Parse result file and compare.
+    # TODO(BEAM-XXXX): taxiride examples should produce int sums, not floats
```
Comment (TheNeuralBit) [inline sdks/python/apache_beam/examples/dataframe/taxiride_test.py:85]: I will file a jira and update these TODOs before merging

**Comment by TheNeuralBit** [comment]
R: @robertwb

**Comment by TheNeuralBit** [comment]
R: @rohdesamuel 
Do you have time to review this?

File: sdks/python/apache_beam/examples/dataframe/taxiride.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+from apache_beam.io import ReadFromText
+from apache_beam.options.pipeline_options import PipelineOptions
+
+ZONE_LOOKUP_PATH = "gs://apache-beam-samples/nyc_taxi/misc/taxi+_zone_lookup.csv"
+
+
+def run_aggregation_pipeline(pipeline_args, input_path, output_path):
+  # The pipeline will be run on exiting the with block.
+  with beam.Pipeline(options=PipelineOptions(pipeline_args)) as p:
+    rides = p | read_csv(input_path)
```
Comment (rohdesamuel) [inline sdks/python/apache_beam/examples/dataframe/taxiride.py:40]: For my own edification, does read_csv expand and return a beam DataFrame?

File: sdks/python/apache_beam/examples/dataframe/taxiride.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      '--output',
+      dest='output',
+      required=True,
+      help='Output file to write results to.')
+  parser.add_argument(
+      '--zone_lookup',
+      dest='zone_lookup_path',
+      default=ZONE_LOOKUP_PATH,
+      help='Location for taxi zone lookup CSV.')
+  parser.add_argument('--pipeline', dest='pipeline', default='location_id_agg')
```
Comment (rohdesamuel) [inline sdks/python/apache_beam/examples/dataframe/taxiride.py:95]: Please also put the different values pipeline can take in the help message.

**Comment by rohdesamuel** [comment]
lgtm just a couple of comments, ty!

**Review by rohdesamuel — APPROVED** [review]

File: sdks/python/apache_beam/examples/dataframe/taxiride.py (hunk 2 above)
Comment (TheNeuralBit) [reply sdks/python/apache_beam/examples/dataframe/taxiride.py:40]: yeah the `read_*` methods generally return `DeferredDataFrame` instances (when applied to a pipeline). If pandas has and read methods that return a Series, we should just produce a `DeferredSeries`. I'm not sure if any readers like that exist though, maybe [read_feather](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_feather.html) can

**Comment by robertwb** [comment]
Just a thought. We have various wordcount examples to demonstrate various ways of doing things. Maybe it makes sense to leave the dataframe one there to inform people about dataframes?

**Comment by TheNeuralBit** [comment]
> Just a thought. We have various wordcount examples to demonstrate various ways of doing things. Maybe it makes sense to leave the dataframe one there to inform people about dataframes?

Fair point. I also like having all the dataframe examples in one module with a README though... 
What if we have both `examples.wordcount_dataframe` and `examples.dataframe.wordcount` and the former just calls out to the latter?

**Comment by TheNeuralBit** [comment]
Ok I added `wordcount_dataframe` back as an alias for `dataframe.wordcount` https://github.com/apache/beam/pull/14308/commits/3fb8fa24b276159b093257a108d9305b4be996a1, LMK if that's acceptable @robertwb

**Comment by robertwb** [comment]
Sounds good.

**Comment by TheNeuralBit** [comment]
GHA failures look like flakes

**Pull request merged**